### PR TITLE
 Feature/rhizome server base

### PR DESCRIPTION
**Pull request by geekbeast**

### CODE REVIEW

**Comment by geekbeast** [comment]
@UnsungHero97 @makosblade Tagged with new minor version number as well to avoid breaking builds.

File: src/main/java/com/geekbeast/rhizome/pods/hazelcast/RegistryBasedHazelcastInstanceConfigurationPod.java [hunk 1]
```diff
@@ -47,231 +30,82 @@
...
+        Preconditions.checkNotNull( mapStore, "Cannot register null map-store." );
         mapRegistry.put( mapName, mapStore );
     }
 
     public static void register( Class<?> hzSerializableClass, Serializer serializer ) {
+        Preconditions.checkNotNull( serializer, "Cannot register null serializer." );
         serializerRegistry.put( hzSerializableClass, serializer );
     }
 
-    // TODO: Make map and serializer beans as startup will fail if there is not at least one bean of that type.
```
Comment (makosblade) [inline src/main/java/com/geekbeast/rhizome/pods/hazelcast/RegistryBasedHazelcastInstanceConfigurationPod.java]: What did we do to not need these anymore?

**Comment by makosblade** [comment]
I think this probably breaks the V2 stuff but I can just fix that on another branch. The class hierarchy changes look excellent.

LGTM

File: src/main/java/com/geekbeast/rhizome/pods/hazelcast/RegistryBasedHazelcastInstanceConfigurationPod.java (hunk 1 above)
Comment (geekbeast) [inline src/main/java/com/geekbeast/rhizome/pods/hazelcast/RegistryBasedHazelcastInstanceConfigurationPod.java]: So the JSR-330 folks had a big argument about whether to allow optional injects and they decided that the "correct" way to to use providers.

Both Spring & Guava implemented their own optional injection, so I switched to the spring version of `@Inject` which is `@Autowired` and set `required=false`

**Pull request merged**